src/Cli/dotnet/commands/dotnet-test/TestCommandParser.cs

```diff
@@ -25,7 +23,7 @@ internal static class TestCommandParser
         public static readonly Option<string> SettingsOption = new ForwardedOption<string>(new string[] { "-s", "--settings" }, LocalizableStrings.CmdSettingsDescription)
         {
             ArgumentHelpName = LocalizableStrings.CmdSettingsFile
-        }.ForwardAsSingle(o => $"-property:VSTestSetting={CommandDirectoryContext.GetFullPath(o)}");
+        }.ForwardAsSingle(o => $"-property:VSTestSetting={SurroundWithDoubleQuotes(CommandDirectoryContext.GetFullPath(o))}");
```

Comment:
I am wondering if we should make a small builder function that would accept the category and the value (e.g. `BuildProperty("VSTestSetting", CommandDirectoryContext.GetFullPath(o))`), in this helper we could build all the common parts like `-property:XXX=YYY` and always escape the entry.

Reply:
Nice! Any preference in doing so in this PR or in a follow-up?

Reply:
`ForwardAsSingleMSBuildProperty(string propertyName,string propertyValue`) or similar?

Reply:
I did a follow-up PR as there are a couple of classes impacted by this refactoring, PR (draft) can be found here #23730.